stdlib/public/core/StringCreate.swift

```diff
@@ -194,6 +209,46 @@ extension String {
     let str = contents.withUnsafeBufferPointer { String._uncheckedFromUTF8($0) }
     return (str, repaired)
   }
+  
+  @usableFromInline @inline(never) // can't be inlined w/out breaking ABI
+  @_specialize(
+    where Input == UnsafeBufferPointer<UInt8>, Encoding == Unicode.ASCII)
+  @_specialize(
+    where Input == Array<UInt8>, Encoding == Unicode.ASCII)
+  internal static func _fromCodeUnits<
+    Input: Collection,
+    Encoding: Unicode.Encoding
+  >(
+    _ input: Input,
+    encoding: Encoding.Type,
+    repair: Bool
+  ) -> (String, repairsMade: Bool)?
+  where Input.Element == Encoding.CodeUnit {
+    guard _fastPath(encoding == Unicode.ASCII.self) else {
+      return _slowFromCodeUnits(input, encoding: encoding, repair: repair)
+    }
+    
+    var result:String? = nil
```

Comment:
`let result: String?` is much cleaner. Then, I don't have to worry about reassignment and I know to look for the definition in both branches because of DI.